lib/fabrics/ofi/src/internal/FILogging.cpp

```diff
@@ -116,11 +114,11 @@ namespace mxl::lib::fabrics::ofi
 
     void fiLog(const struct fi_provider* prov, enum fi_log_level level, enum fi_log_subsys subsys, char const*, int line, char const* msgIn)
     {
-        std::string msg{msgIn};
+        auto msg = std::string{msgIn};
 
-        if (!msg.empty() && msg[msg.size() - 1] == '\n')
+        if (!msg.empty() && (msg[msg.size() - 1] == '\n'))
```

Comment:
Why not also use `-... && (msg.back() == ' \n')`? Maybe I forgot to mention that in my last comment.

TBH: C++20 also has `ends_with('\n')`, that's probably an even better choice.